src/editors/_shared/hooks/DataTypes/types/NumberType.js

```diff
@@ -5,60 +5,19 @@ import DataType from "../AbstractDataType";
 import { checkIfDefaultOrDisabled } from "./utils";
 
 class NumberType extends DataType {
-  // Number type properties definition
   key = DATA_TYPES.NUMBER;
+  inputType = DATA_TYPES.NUMBER;
   label = "Number";
-  default = "0";
+  default = 0;
 
-  editComponent = props => {
-    return (
-      <TextField
-        fullWidth
-        type={DATA_TYPES.NUMBER}
-        placeholder={"0"}
-        value={props.rowData.value || ""}
-        disabled={props.disabled}
-        onChange={evt => props.onChange(evt.target.value)}
-      ></TextField>
-    );
-  };
-
-  /**
-   * Validate number value
-   * @param {*} value
-   * @returns
-   */
-  validate(value) {
-    return new Promise(resolve => {
-      try {
-        if (checkIfDefaultOrDisabled(value)) {
-          return resolve({ success: true, value });
-        }
-        const parsed = this.parseValueToFloat(value);
-        const isValid = typeof parsed === DATA_TYPES.NUMBER && !isNaN(parsed);
-        resolve({ success: isValid, parsed });
-      } catch (e) {
-        resolve({ success: false });
-      }
-    });
+  _validate(value) {
+    return value === undefined
+      || value !== null && typeof value === this.key
+      && !isNaN(value) && !Array.isArray(value);
```

Comment:
Mixing operators without a clear goal is always dangerous. 
Is this what you want?

```suggestion
    return value === undefined || 
      (value !== null && typeof value === this.key
      && !isNaN(value) && !Array.isArray(value));
```

Reply:
I've replied below

Reply:
In other words, the goal is clear, the understanding might not be.